app/services/zendesk/client.rb

```diff
@@ -0,0 +1,16 @@
+module Zendesk
+  class Client
+    include Singleton
```

Comment:
This will deal with the concurrency issue raised a while back

Reply:
what was raised a while back?